### DISCUSSION TITLE
Shell

### DISCUSSION THREAD

**Discussion by jgomez720** [KittyCAD/modeling-app#503] [New Feature Requests]
### Description
A tool that enables the user to create a shell of a body. This essentially means making the part hollow, except for one side is completely open (unless the user specifies "hollow" in the options).

### Functionality
1. The user selects the _shell_ button.
2. The user selects the face they want to shell.
3. The user inputs the wall thickness dimension.
4. Confirm.

OR

1. The user selects the face they want to shell.
2. The user selects the _shell_ button.
3. The user inputs the wall thickness dimension.
4. Confirm.

### Supporting Docs/Images
![Screenshot 2023-11-08 at 2 35 35 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/2a2dc5c3-be82-4cac-93c9-069e3c5cbb56)
![Screenshot 2023-11-08 at 2 35 49 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/90d7f280-21c1-41eb-8ecf-7f5e32e91796)
![Screenshot 2023-11-08 at 2 35 56 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/2c2a0287-b446-4b6e-b221-7d2d5b629820)

## PseudoKCL
```
const firstSketch = startSketchOn('XY')
  |> startProfileAt([-12, 12], %)
  |> line([24, 0], %)
  |> line([0, -24], %)
  |> line([-24, 0], %)
  |> close(%)
  |> extrude(6, %)

shell({
  faces: firstSketch.extrudeEnd(), // This is just my way of saying "grab the face at the end when firstSketch is extruded". Very certain there's something better.
  thickness: 0.25,
})
```
**Result:**
![Screenshot 2023-11-09 at 10 59 33 AM](https://github.com/KittyCAD/modeling-app/assets/114548659/795ee520-7b92-485d-b9af-461c67121483)

## AdvancedKCL
```
const firstSketch = startSketchOn('XY')
  |> startProfileAt([-12, 12], %)
  |> line([24, 0], %)
  |> line([0, -24], tag: 'shellface1', %)
  |> line([-24, 0], tag: 'shellface2', %)
  |> close(%)
  |> extrude(6, %)

shell({
  faces: [firstSketch.extrudeEnd(), 'shellface1'.extrudeFace(), 'shellface2'.extrudeFace()], // Same as above, but with additions. The reason I'm putting .extrudeFace() at the end of the tags is because this differentiates grabbing the edge vs the face. If I just use the tag itself, I'm saying to use the edge itself.
  thickness: 0.25,
})


```
**Result:**
![Screenshot 2023-11-09 at 11 00 14 AM](https://github.com/KittyCAD/modeling-app/assets/114548659/30684ef8-4cd2-4a2d-9e84-59333b0a3939)

**Comment by jessfraz** [comment]
closed by https://github.com/KittyCAD/modeling-app/pull/2683
